modules/openapi-generator/src/test/resources/3_0/oneOfDiscriminator.yaml

```diff
@@ -170,6 +170,37 @@ components:
           type: integer
       oneOf:
         - $ref: '#/components/schemas/FruitType'
+    FruitTypeEnum:
+      type: string
+      enum: [APPLE, BANANA]
+    FruitOneOfEnumMappingDisc:
+      type: object
+      properties:
+        fruitType:
+          $ref: "#/components/schemas/FruitTypeEnum"
+      required: fruitType
+      oneOf:
+        - $ref: '#/components/schemas/AppleOneOfEnumMappingDisc'
+        - $ref: '#/components/schemas/BananaOneOfEnumMappingDisc'
+      discriminator:
+        propertyName: fruitType
+        mapping:
```

Comment:
The mapping to the enum value might cause problems with the auto generated JsonSubType.

Reply:
I actually generated a sample Project:

```
@JsonIgnoreProperties(
  value = "fruitType", // ignore manually set fruitType, it will be automatically generated by Jackson during serialization
  allowSetters = true // allows the fruitType to be set during deserialization
)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "fruitType", visible = true)
@JsonSubTypes({
  @JsonSubTypes.Type(value = AppleOneOfEnumMappingDisc.class, name = "APPLE"),
  @JsonSubTypes.Type(value = AppleOneOfEnumMappingDisc.class, name = "AppleOneOfEnumMappingDisc"),
  @JsonSubTypes.Type(value = BananaOneOfEnumMappingDisc.class, name = "BANANA"),
  @JsonSubTypes.Type(value = BananaOneOfEnumMappingDisc.class, name = "BananaOneOfEnumMappingDisc")
})

@Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2022-11-03T09:30:55.032940+01:00[Europe/Berlin]")
public interface FruitOneOfEnumMappingDisc {
    public FruitTypeEnum getFruitType();
}

```

@welshm Isn't there another issue / pr dealing with duplicated JsonSubTypes?

Reply:
@bernie-schelberg-mywave Could you please add a enum based discriminator to Ione of the generated sample projects (for example `bin/configs/spring-boot-oneof.yaml`)?

Reply:
> I actually generated a sample Project:
> 
> ```
> @JsonIgnoreProperties(
>   value = "fruitType", // ignore manually set fruitType, it will be automatically generated by Jackson during serialization
>   allowSetters = true // allows the fruitType to be set during deserialization
> )
> @JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "fruitType", visible = true)
> @JsonSubTypes({
>   @JsonSubTypes.Type(value = AppleOneOfEnumMappingDisc.class, name = "APPLE"),
>   @JsonSubTypes.Type(value = AppleOneOfEnumMappingDisc.class, name = "AppleOneOfEnumMappingDisc"),
>   @JsonSubTypes.Type(value = BananaOneOfEnumMappingDisc.class, name = "BANANA"),
>   @JsonSubTypes.Type(value = BananaOneOfEnumMappingDisc.class, name = "BananaOneOfEnumMappingDisc")
> })
> 
> @Generated(value = "org.openapitools.codegen.languages.SpringCodegen", date = "2022-11-03T09:30:55.032940+01:00[Europe/Berlin]")
> public interface FruitOneOfEnumMappingDisc {
>     public FruitTypeEnum getFruitType();
> }
> ```
> 
> @welshm Isn't there another issue / pr dealing with duplicated JsonSubTypes?

I'm running openapi-generator version 6.2.0 and the above is the current behaviour when you add a mapping to the discriminator. Both the derived classes and the mapping values are added as JsonSubTypes.  

The documentations seems to indicate that this is the expected behaviour. 

> The mapping in the discriminator includes any descendent schemas that allOf inherit from self, any oneOf schemas, any anyOf schemas, any x-discriminator-values, and the discriminator mapping schemas in the OAS document AND Codegen validates that oneOf and anyOf schemas contain the required discriminator and throws an error if the discriminator is missing.

Reply:
@cachescrubber Yes, there's another project for duplicated JsonSubTypes: https://github.com/OpenAPITools/openapi-generator/pull/13815